### PR TITLE
nginx router based on template

### PR DESCRIPTION
**Pull request by rajatchopra**
template based nginx router implementation

/cc @jawnsy

### CODE REVIEW

**Comment by smarterclayton** [comment]
Nice

**Comment by louyihua** [comment]
nginx supports include conf files, so it maybe better to split routes into individual conf files, one file per route.

**Comment by rajatchopra** [comment]
@openshift/networking Review please

File: images/router/nginx/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#
+# This is the NGINX router for OpenShift Origin.
+#
+# The standard name for this image is openshift/origin-nginx-router
+#
+FROM openshift/origin
+
+RUN INSTALL_PKGS="nginx" && \
```
Comment (pecameron) [inline images/router/nginx/Dockerfile]: @rajatchopra Just curious here. How do we specify which version of nginx is used?

Comment (rajatchopra) [reply images/router/nginx/Dockerfile]: Not specify-able yet. This is the base image being built. And we will pick the standard install from openshift/origin image's configured repositories.

**Comment by lihongan** [comment]
@rajatchopra How to start the nginx router ? Does it need other special options or same to haproxy router? e.g. just using `oadm router --images=registry/openshift3/ose-nginx-router`

File: images/router/nginx/reload-nginx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -o nounset
+
+config_file=/var/lib/nginx/conf/nginx.config
+if [ -f /var/lib/nginx/logs/nginx.pid ]; then
+/usr/sbin/nginx -c ${config_file} -s reload
```
Comment (lihongan) [inline images/router/nginx/reload-nginx]: indent ?

**Comment by rajatchopra** [comment]
@lihongan 
Yes start with 'oadm router --images=openshift/nginx-router', but we need to edit the dc for working-dir also. So the deployment config should have:

```
 - images: ...
 - command:
 - /usr/bin/openshift-router
 - --loglevel=2
 - --working-dir=/var/lib/nginx/router
```


> How to start the nginx router ? Does it need other special options or same to haproxy router? e.g. just using oadm router --images=registry/openshift3/ose-nginx-router

**Comment by rajatchopra** [comment]
/assign @smarterclayton 
Clayton, with this change the nginx router image will also be built. Do we need a 'community'/'contrib'/'vendor' kind of setup for such images?  At least until this is fully supported.

**Comment by rajatchopra** [comment]
@eparis Removed the default pub keys file (was initially meant to provide a default cert for tls connections).

**Comment by smarterclayton** [comment]
No it's fine to add it.  Did you add the push rules to hack/push-images?

**Pull request merged**